src/cli/test/parse-flags.spec.ts

```diff
@@ -160,12 +161,41 @@ describe('parseFlags', () => {
   it('should parse --ci', () => {
     args[0] = '--ci';
     const flags = parseFlags(args, sys);
+    expect(flags.knownArgs).toEqual(['--ci']);
     expect(flags.ci).toBe(true);
   });
 
+  describe.each(BOOLEAN_CLI_ARGS)('should parse boolean flag %s', (cliArg) => {
```

Comment:
these comprehensive tests of all the supported args serve as regression tests against duplicating any of the arguments in the arrays - because of the way that the arg parsing algorithm works having a dupe will result in a value like `[true, true]` being set on `ConfigFlags`. We want to make sure that can't happen, so we have some simple tests here.